microsoft-edge/progressive-web-apps-chromium/how-to/best-practices.md

```diff
@@ -0,0 +1,205 @@
+
+#### Shortcuts
+
+Shortcuts make it possible to define direct links to your app functionality by right-clicking the app icon.
+
+See [Define app shortcuts](./shortcuts.md).
+
+#### App sharing
+
+PWAs can share content to other apps, and receive content from other apps, via the operating system's share dialog.
+
+See [Share content with other apps](./share.md).
+
+#### Icon badges
+
+PWAs can display status badges on their app icons to alert users of the presence of new content in the app.
+
+See [Display a badge on the app icon](./notifications-badges.md#display-a-badge-on-the-app-icon) in _Re-engage users with badges, notifications, and push messages_.
+
+#### File handling
+
+You can register your PWA to be a file handler for certain file types.
+
+Use the PWA file handler feature to automatically launch your app when certain file are opened by the user in the operating system.
+
+See [Handle files in Progressive Web Apps](./handle-files.md).
+
+#### Link and protocol handling
+
+Register your app is a link or protocol handler to automatically launch your app when certain links are used in other applications.
+
+See [Handle links to your Progressive Web Apps](./handle-urls.md) and [Handle protocols in Progressive Web Apps](./handle-protocols.md).
+
+<!-- 
+#### Widgets
+
+TODO: when widgets doc land, add a section about it here.
+ -->
+
+<!-- ====================================================================== -->
+## Support offline scenarios
+
+When connected to a slow or unreliable network, or even with no internet access, installed applications can usually still be opened and used. Users of installed applications expect them to continue working under these conditions. They also expect the network-dependent parts of an application to gracefully handle the lack of connection with a custom message and local caching capabilities.
+
+To improve retention, use the service worker's `Fetch` and `Cache` APIs, and local data storage access and provide a good offline experience to your users.
+
+You can provide a good offline experience in several steps:
+
+1. Cache all the static resources your app needs to start, display content, and let users perform the most common tasks.
+1. Redirect to a custom offline page for the pages that can't be cached.
+1. Gracefully handle the tasks of your app that can't be used with no internet access.
+1. Implement advanced scenarios such as letting users download data for offline viewing.
+
+To learn more, see [Offline and network connectivity support in Progressive Web Apps](./offline.md).
+
+
+<!-- ====================================================================== -->
+## Store data locally
+
+To provide a rich and fast experience to your users, use the different persistent data storage options available to PWAs, such as:
```

Comment:
```suggestion
To provide a rich and fast experience to your users, use the different persistent data storage options that are available to PWAs, such as:
```
I think Style Guide says to include clarifying optional words more than telegraphic abbreviated style.